bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,377 @@
+# Importing everything
+import asyncio
+import json
+import logging
+import random
+from calendar import month_name
+from collections import defaultdict
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, Union
+
+from async_rediscache import RedisCache
+from discord import Embed, Message, Reaction, TextChannel, User
+from discord.ext import tasks
+from discord.ext.commands import Bot, Cog, Context, group
+
+from bot.constants import Channels, Client, Colours, Month
+from bot.utils.decorators import InMonthCheckFailure
+
+logger = logging.getLogger(__name__)
+MONTH_TO_LIMIT_TO = Month.OCTOBER
+
+EMOJIS_VAL = {  # the value of the emojis
+    '\N{Jack-O-Lantern}': 1,
+    '\N{Ghost}': 2,
+    '\N{Skull and Crossbones}': 3,
+    '\N{Zombie}': 4,
+    '\N{Face Screaming In Fear}': 5
+}
+
+
+class SpookNameRate(Cog):
+    """
+    A game that asks the user to spookify or halloweenify a name that is given everyday.
+
+    It sends a random name everyday. The user needs to try and spookify it to his best ability and
+    send that name back using the `spooknamerate add entry` command
+    """
+
+    messages = RedisCache()  # a variable to store all the messages
+    data = RedisCache()
```

Comment:
Above each of these caches, please explain more what they store and in what format.